### DISCUSSION TITLE
Is there a way to use nats.EncodedConn as connection for micro.Service?

### DISCUSSION THREAD

**Discussion by Archirk** [nats-io/nats.go#1481] [Q&A] [answered]
I have handlers `func(subject, reply string, obj proto.Message)`  and i wish i could move from my bad abstractions to nats abstraction of `micro.Service`.  
However `micro.AddService()` can not accept `nats.EncodedConn` as connection.  
I also do not see any options in `micro.Config` or `micro.EndpointConfig` to set somehow encoder.  
Do I miss something? Why encoded connection can not be used as connection for service?  Is there a good way around it?

**Comment by ripienaar** [comment, marked as answer]
We're basically considering EncodedConn as not something we'll take forward but rather try and find a better option.

It's best to avoid it for the moment in general - micro and otherwise.

**Reply by Archirk** [reply]
Could you advice a "best practice" to parse protoData into go-struct to avoid doing it in each handler? Is there the "middleware" in case on nats lib?

**Reply by ripienaar** [reply]
Not really no - but a helper function taking protobytes and a prepared proto message would be good - however given that decoding a proto is like a one liner + error handling I dont think it would be a big gain.
